va/va_trace.c

```diff
@@ -1140,6 +1141,14 @@ static void va_TraceSurfaceAttributes(
     if (!attrib_list || !num_attribs)
         return;
 
+    int32_t memtype = 0;
+    for (i = 0; i < *num_attribs; i++) {
+        if(attrib_list[i].type == VASurfaceAttribMemoryType) {
+            memtype = attrib_list[i].value.value.i;
```

Comment:
Overall this looks like a good fix, which should be a separate MR.

Few small suggestions:
 - seems like the memtype assignment should happen in the loop below, where it's being checked
 - why do we handle mem_type 0, the same as TYPE_DRM_PRIME - that doesn't seem right
 - can we bribe you to add support for DRM_PRIME_2 - thanks o/

Reply:
Thanks. I'd prefer if possible, to keep the fix in this MR, as a separate commit, which I think will be rebase-merged into master, keeping history clean too eventually. By having the fix commit in this MR, I can also then build upon it and add the win32 buffer types in a subsequent commit, and I can avoid the crash when I test `vaExportSurfaceHandle`. As this MR targets `va_win` branch, it'd be easier to avoid rebasing from multiple branches once the fix goes into master in a separate PR.

Regarding your other points I've added some comment in the code to clarify:
1) I'm preferred iterating over the attributes first, in case the API client sends `VASurfaceAttribMemoryType`  before `VASurfaceAttribExternalBufferDescriptor` in `attrib_list` as I didn't find any order specification in the docs, does this make sense ?
2) I found a note in va.h stating that `VASurfaceAttribExternalBuffers` type should be assumed for `VASurfaceAttribExternalBufferDescriptor` unless the memory type is provided.
3) I've added support for printing `VADRMPRIMESurfaceDescriptor` in the commit along with the fix, and then added a new commit with win32 types trace support too.

Reply:
Makes sense, thanks for going the extra mile with DRM_PRIME_2 :heart: